src/adaptations/device-layer/include/Weave/DeviceLayer/OpenThread/GenericThreadStackManagerImpl_OpenThread.ipp

```diff
@@ -150,15 +150,24 @@ void GenericThreadStackManagerImpl_OpenThread<ImplClass>::_OnPlatformEvent(const
 {
     if (event->Type == DeviceEventType::kThreadStateChange)
     {
-#if WEAVE_DETAIL_LOGGING
-
         Impl()->LockThreadStack();
 
+#if WEAVE_DETAIL_LOGGING
+
         LogOpenThreadStateChange(mOTInst, event->ThreadStateChange.OpenThread.Flags);
 
-        Impl()->UnlockThreadStack();
+        // The API we provide to enable/disable thread controls both the Thread
+        // stack and the IP6 stack within OpenThread.  In order to properly
+        // initialize OpenThread, we need to enable IP6 in OT, and we can
+        // disable it once the OT stack is up and running and signaling events.
+        if ((otThreadGetDeviceRole(mOTInst) == OT_DEVICE_ROLE_DISABLED) && otIp6IsEnabled(mOTInst))
+        {
+            otIp6SetEnabled(mOTInst, false);
+        }
 
 #endif // WEAVE_DETAIL_LOGGING
```

Comment:
Should move 
`#endif // WEAVE_DETAIL_LOGGING`
after line 158 otherwise the check in line 163 will get disabled if WEAVE_DETAIL_LOGGING is turned off.

Reply:
Yup, you're right. fixed.